Tribler/community/tunnel/conversion.py

```diff
@@ -58,13 +85,13 @@ def _decode_cell(self, placeholder, offset, data):
 
     def _encode_create(self, message):
         payload = message.payload
-        packet = pack("!IHH20s", payload.circuit_id, len(payload.node_public_key),
-                      len(payload.key), payload.node_id) + payload.node_public_key + payload.key
+        packet = pack("!IHH20s?", payload.circuit_id, len(payload.node_public_key),
+                      len(payload.key), payload.node_id, payload.exit_candidates) + payload.node_public_key + payload.key
```

Comment:
Node n-1 still does not know what n (length of the circuit) is and what data is exited over the circuit. To achieve the same level of anonimity the circuit length should be at least 3 hops to prevent node n-1 from knowing both its predecessor and succesor (in the case of 2 hops node n-1 knows both the exit-node and initator, that's unsafe). 

I don't think there is a way to prevent this leaking, because the last hop is always responsible of returning a list of candidates. Btw, it is a temporary solution for anonymous downloading. When hidden seeding will become the standard in Tribler, exit-nodes are not relevant anymore.